python/lsst/ts/ess/csc/rpi_data_client.py

```diff
@@ -350,21 +359,23 @@ async def run(self) -> None:
                 else:
                     self.log.warning(f"Ignoring unparsable data: {data}.")
 
-            err_msg = "Connection lost."
-            self.log.error(err_msg)
-            raise ConnectionError("Connection lost")
-        except asyncio.CancelledError:
-            self.log.debug("read_loop cancelled")
-            raise
-        except asyncio.TimeoutError:
-            self.log.error("Read timed out")
-            raise
-        except RuntimeError as e:
-            self.log.error(f"read_loop failed: {e}")
-            raise
-        except Exception:
-            self.log.exception("read_loop failed")
-            raise
+            except asyncio.CancelledError:
+                self.log.debug("read_loop cancelled")
+                raise
+            except asyncio.TimeoutError:
+                self.log.warning("Read timed out")
```

Comment:
I strongly suggest adding the number of consecutive timeouts to this log message. That will make the message far more useful, since it will be easier to see the difference between "the system is probably down" (the number keeps going up) vs "the system is flakey" (the message is sporadic and the value is usually 1).

Reply:
Yes I agree. Done.